### PR TITLE
Shadows 2

### PR DESCRIPTION
**Pull request by 1whatleytay**
Rebased up to date version of #580, everything's bent's. He asked me to open a new PR for him so here it is.

Will fix the commit history in a bit.

### CODE REVIEW

**Comment by Zangetsu38** [comment]
Last commit have fail on format ^^

**Comment by 1whatleytay** [comment]
Sorry not sure what you mean,..

File: vita3k/modules/SceGxm/SceGxm.cpp [hunk 1]
```diff
@@ -663,6 +671,7 @@ static int SDLCALL thread_function(void *data) {
         if (!display_callback)
             break;
 
+		SceGxmSyncObject *oldBuffer = Ptr<SceGxmSyncObject>(display_callback->old_buffer).get(*params.mem);
```
Comment (Zangetsu38) [inline vita3k/modules/SceGxm/SceGxm.cpp]: this

Comment (1whatleytay) [reply vita3k/modules/SceGxm/SceGxm.cpp]: thanks!

**Comment by 1whatleytay** [comment]
Conflicts fixed + weird tabbing.

Do you mind if we keep the commit structure? I want to get this merged today or tomorrow.

**Comment by VelocityRa** [comment]
Fine by me. Let's get this merged finally :)

**Pull request merged**